src/figuretype/trader.c

```diff
@@ -1036,16 +1036,21 @@ int figure_trade_sea_trade_units()
 
 int figure_trader_ship_docked_once_at_dock(figure *ship, int dock_id)
 {
-    for (int i = 0; i < 10; i++) {
+    for (int i = 0; i < MAX_DOCKS; i++) {
         if (dock_id == city_buildings_get_working_dock(i)) {
-            if (ship->building_id & 1 << i) {
+            if (figure_trader_ship_docked_once_at_dock_by_num(ship, i)) {
                 return 1;
             }
         }
     }
     return 0;
 }
 
+int figure_trader_ship_docked_once_at_dock_by_num(figure *ship, int dock_num)
```

Comment:
I think `figure_trader_ship_already_docked_at` would be more concise and would work as well.